manifests/params.pp

```diff
@@ -17,21 +17,29 @@
   $version = 'latest'
   $service = 'jenkins'
   $install_from_git = false
+  $install_from_pip = true
+  $install_from_pkg = false
   $git_revision     = 'master'
   $git_url          = 'https://git.openstack.org/openstack-infra/jenkins-job-builder'
 
   case $::osfamily {
     'RedHat', 'Amazon': {
-      $python_packages = [ 'python', 'python-devel', 'python-pip', 'python-argparse']
+      if $::operatingsystemrelease =~ /^6/ {
+        $python_packages = [ 'python', 'python-devel', 'python-pip', 'python-argparse', 'PyYAML']
+        # This requires the dcaro/jenkins-job-builder repository
+        $jjb_packages    = [ 'jenkins-job-builder']
+      } else {
+        $python_packages = [ 'python', 'python-devel', 'python-pip', 'PyYAML']
```

Comment:
> ya.. this update aimed to make all of that more consistent because some ppl (like myself) have rules against installing software not from system packages. hopefully that helps

Yeah, that's fair enough really. I changed our resource chaining, and it all looks to be working as expected ... Just a few more real world examples to run, and I think it should be good to go! Will let you know when done - should be very soon.